modules/solid_mechanics/src/vectorpostprocessors/CrackFrontNonlocalStress.C

```diff
@@ -0,0 +1,165 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "CrackFrontNonlocalStress.h"
+#include "Moose.h"
+#include "MooseEnum.h"
+#include "RankTwoTensor.h"
+#include "Conversion.h"
+#include "libmesh/string_to_enum.h"
+#include "libmesh/quadrature.h"
+#include "libmesh/utility.h"
+#include "CrackFrontDefinition.h"
+#include "RankTwoScalarTools.h"
+#include "libmesh/utility.h"
+
+registerMooseObject("SolidMechanicsApp", CrackFrontNonlocalStress);
+
+InputParameters
+CrackFrontNonlocalStress::validParams()
+{
+  InputParameters params = ElementVectorPostprocessor::validParams();
+  params.addRequiredParam<UserObjectName>("crack_front_definition",
+                                          "The CrackFrontDefinition user object name");
+  params.addRequiredParam<Real>("box_length", "Distance in front of crack front.");
+  params.addRequiredParam<Real>("box_height", "Distance normal to front of crack front.");
+  params.addParam<Real>("box_width", 1, "Distance tangent to front of crack front.");
```

Comment:
Weren't we going to have this default to something based on the segment length? Regardless, a default of 1 probably isn't a good idea here.

Reply:
Thanks, I missed where the error was generated. I think the best way to handle this is to just not set the default of 1 in line 32. It doesn't really make much (maybe any) difference in the way the code runs, but it wouldn't print 1 as a default value in the documentation. There's not a case where you would fall back on using 1, so it's not meaningful to set a default value.